### ISSUE TITLE
Creation of cloudflare_origin_ca_certificate resource using API_TOKEN

### ISSUE THREAD

**Issue by saveler** [cloudflare/terraform-provider-cloudflare#2116]
### Current Terraform and Cloudflare provider version

```
Terraform v1.3.2
Cloudflare Provider 3.20.0
```

### Description

Right now in _cloudflare_origin_ca_certificate_ resource we can see this NOTE:
`This resource requires you use your Origin CA Key as the [api_user_service_key](https://registry.terraform.io/providers/cloudflare/cloudflare/latest/docs/index#api_user_service_key) value.`

Since August of 2022 it is possible to use API_TOKEN to request origin certificate. (It is confirmed by Cloudflare support, the same info can be found in CF API docs https://developers.cloudflare.com/api/operations/origin-ca-create-certificate)
For us it is important to use API_TOKEN to get Origin Certificate due to our permission model in our company. We also found a lot of github issues related to it, so we expect that a lot of users want to see this feature.

Please, add possibility to use API_TOKEN with _cloudflare_origin_ca_certificate_ resource.

**Proof that API_TOKEN can be used:
Request:**
`root@6d94198:/testme# curl -X POST "https://api.cloudflare.com/client/v4/certificates" -H "Authorization: Bearer ***" --data '{"hostnames":["****"],"requested_validity":730,"request_type":"origin-rsa","csr":"-----BEGIN CERTIFICATE REQUEST-----\n***-----END CERTIFICATE REQUEST-----\n"}'`

**Response:**
`{"success":true,"result":{"id":"***","certificate":"-----BEGIN CERTIFICATE-----\n***\n-----END CERTIFICATE-----\n","expires_on":"2025-01-01 13:03:00 +0000 UTC","request_type":"origin-rsa","hostnames":[***],"csr":"-----BEGIN CERTIFICATE REQUEST-----\n****\n-----END CERTIFICATE REQUEST-----\n","requested_validity":730},"errors":[],"messages":[]}
`

### Use cases

If I'm not mistaken, right now there is only one resource that requires `api_user_service_key`: `cloudflare_origin_ca_certificate`
After this change it won't be required anymore to pass `api_user_service_key` in order to create `cloudflare_origin_ca_certificate` and the same provider credentials will be used for all the resources.

### Potential Terraform configuration

```
provider "cloudflare" {
  api_token = "***"
}

resource "tls_private_key" "example" {
  algorithm = "RSA"
}

resource "tls_cert_request" "example" {
  key_algorithm   = tls_private_key.example.algorithm
  private_key_pem = tls_private_key.example.private_key_pem

  subject {
    common_name  = ""
    organization = "Terraform Test"
  }
}

resource "cloudflare_origin_ca_certificate" "example" {
  csr                = tls_cert_request.example.cert_request_pem
  hostnames          = ["example.com"]
  request_type       = "origin-rsa"
  requested_validity = 7
}
```

**Comment by jacobbednarz** [comment]
Fixed by cloudflare/cloudflare-go#1161

**Comment by saveler** [comment]
Thank you very much!)
